hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/response/key/OMAllocateBlockResponse.java

```diff
@@ -0,0 +1,59 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hadoop.ozone.om.response.key;
+
+import org.apache.hadoop.ozone.om.OMMetadataManager;
+import org.apache.hadoop.ozone.om.helpers.OmKeyInfo;
+import org.apache.hadoop.ozone.om.response.OMClientResponse;
+import org.apache.hadoop.ozone.protocol.proto.OzoneManagerProtocolProtos;
+import org.apache.hadoop.ozone.protocol.proto.OzoneManagerProtocolProtos
+    .OMResponse;
+import org.apache.hadoop.utils.db.BatchOperation;
+
+import java.io.IOException;
+
+/**
+ * Response for AllocateBlock request.
+ */
+public class OMAllocateBlockResponse extends OMClientResponse {
+
+  private final OmKeyInfo omKeyInfo;
+  private final long clientID;
+
+  public OMAllocateBlockResponse(OmKeyInfo omKeyInfo,
+      long clientID, OMResponse omResponse) {
+    super(omResponse);
+    this.omKeyInfo = omKeyInfo;
+    this.clientID = clientID;
+  }
+
+  @Override
+  public void addToDBBatch(OMMetadataManager omMetadataManager,
+      BatchOperation batchOperation) throws IOException {
+
+    // For OmResponse with failure, this should do nothing. This method is
+    // not called in failure scenario in OM code.
+    if (getOMResponse().getStatus() == OzoneManagerProtocolProtos.Status.OK) {
```

Comment:
This check is performed in all OMClientResponse classes. We can move this check to a calling function.
Again, this can be done in a different Jira if thats preferred.

Reply:
This is actually a safe guard check. As these now each request/response classes are independent. If some one in the future directly uses these classes (Don't know if someone will use them) to handle those cases added the check. Right now when OMResponse Status is OK, then only response is added to OM DoubleBuffer in OzoneManagerProtocolServerSideTranslatorPB.java. And this addToDbBatch will be called from OMDoubleBuffer. So, for now, even if someone in future who implements these classes forgot to add this check, we don't see any side affects.

Reply:
In future if some is calling these classes from somewhere else, they should take care of it. We cannot future proof code fully anyway :)

Reply:
If someone uses these classes and have OMResponse with error, we have guarded them. But someone writes the classes without this check, and then someone uses these classes we shall be in problem. But in our scenario with the current code, it will not be an issue.

On a side note, the reason for this check is when the response is an error, all other parameters are set to null in response classes. This is the main reason for the check.